File: Languages/Java/RandAvg.java
import java.io.*;
import java.util.*;
import java.nio.file.*;

public class RandAvg {
    
    public static void main(String[] args) {
        System.out.println("Hello, World!");
        generateRandomNumbers();
    }
    
    private static void generateRandomNumbers() {
        List<Integer> randomNumbers = new ArrayList<>();
        final int count = 1000;
        final String outputDir = "../rand_avg output";
        final String outputFile = outputDir + "/random_numbers.txt";
        
        // Create output directory
        try {
            Files.createDirectories(Paths.get(outputDir));
        } catch (IOException e) {
            System.out.println("Could not create output directory");
            return;
        }
        
        // Random number generator
        Random random = new Random();
        
        // Generate 1000 random numbers
        for (int i = 0; i < count; i++) {
            randomNumbers.add(random.nextInt(1000));
        }
        
        // Calculate mean
        double sum = 0;
        for (int num : randomNumbers) {
            sum += num;
        }
        double mean = sum / count;
        
        // Save to file
        try (PrintWriter writer = new PrintWriter(new FileWriter(outputFile))) {
            for (int num : randomNumbers) {
                writer.println(num);
            }
        } catch (IOException e) {
            System.out.println("Could not create output file");
            return;
        }
        
        System.out.println("Generated 1000 random numbers");
        System.out.printf("Mean: %.2f%n", mean);
        System.out.println("Saved to: " + outputFile);
    }
}

File: Languages/Kotlin/rand_avg.kt
import java.io.File
import kotlin.random.Random

// Hello, World!
fun hello() {
    println("Hello, World!")
}

fun generateRandomNumbers() {
    val count = 1000
    val outputDir = "../rand_avg output"
    val outputFile = "$outputDir/random_numbers.txt"
    
    // Create output directory
    File(outputDir).mkdirs()
    
    // Generate 1000 random numbers
    val randomNumbers = mutableListOf<Int>()
    var sum = 0L
    
    repeat(count) {
        val num = Random.nextInt(0, 1000)
        randomNumbers.add(num)
        sum += num
    }
    
    // Calculate mean
    val mean = sum.toDouble() / count
    
    // Save to file
    try {
        File(outputFile).writeText(randomNumbers.joinToString("\n"))
        
        println("Generated 1000 random numbers")
        println("Mean: ${String.format("%.2f", mean)}")
        println("Saved to: $outputFile")
    } catch (e: Exception) {
        println("Could not create output file")
    }
}

fun main() {
    hello()
    generateRandomNumbers()
}
